pkg/apis/pipeline/v1beta1/pipeline_validation_test.go

```diff
@@ -2571,6 +2572,72 @@ func TestPipelineTasksExecutionStatus(t *testing.T) {
 	}
 }
 
+// TestIncompatibleAPIVersions exercises validation of fields that
+// require a specific feature gate version in order to work.
+func TestIncompatibleAPIVersions(t *testing.T) {
```

Comment:
NIT: this is testing just the matrix, suggest being explicit in the name 🙏

Reply:
was thinking that we can support checking other alpha features in pipelines here, but makes sense to limit it to matrix for now until we have more alpha features in pipeline spec